src/io.c

```diff
@@ -24,6 +24,10 @@
 #define DISPATCH_IO_DEBUG DISPATCH_DEBUG
 #endif
 
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 4096
```

Comment:
then you should use getpagesize() or sysconf(_SC_PAGESIZE) for the define. I assume arm doesn't define it because there are both 4k and 16k devices out there and it's not tied to the ABI.